src/coredata.cs

```diff
@@ -1140,23 +1312,147 @@ public interface NSPersistentStore {
+		bool ShouldInferMappingModelAutomatically { get; set; }
+
+		// -(instancetype _Nonnull)initWithURL:(NSURL * _Nonnull)url __attribute__((objc_designated_initializer));
+		[Export ("initWithURL:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSUrl url);
+	}
+	
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(NSObject))]
+	public interface NSPersistentContainer
+	{
+		// +(instancetype _Nonnull)persistentContainerWithName:(NSString * _Nonnull)name;
+		[Static]
+		[Export ("persistentContainerWithName:")]
+		NSPersistentContainer PersistentContainerWithName (string name);
+
+		// +(instancetype _Nonnull)persistentContainerWithName:(NSString * _Nonnull)name managedObjectModel:(NSManagedObjectModel * _Nonnull)model;
+		[Static]
+		[Export ("persistentContainerWithName:managedObjectModel:")]
+		NSPersistentContainer PersistentContainerWithName (string name, NSManagedObjectModel model);
+
+		// +(NSURL * _Nonnull)defaultDirectoryURL;
+		[Static]
+		[Export ("defaultDirectoryURL")]
+		NSUrl DefaultDirectoryUrl { get; }
+
+		// @property (readonly, copy) NSString * _Nonnull name;
+		[Export ("name")]
+		string Name { get; }
+
+		// @property (readonly, strong) NSManagedObjectContext * _Nonnull viewContext;
+		[Export ("viewContext", ArgumentSemantic.Strong)]
+		NSManagedObjectContext ViewContext { get; }
+
+		// @property (readonly, strong) NSManagedObjectModel * _Nonnull managedObjectModel;
+		[Export ("managedObjectModel", ArgumentSemantic.Strong)]
+		NSManagedObjectModel ManagedObjectModel { get; }
+
+		// @property (readonly, strong) NSPersistentStoreCoordinator * _Nonnull persistentStoreCoordinator;
+		[Export ("persistentStoreCoordinator", ArgumentSemantic.Strong)]
+		NSPersistentStoreCoordinator PersistentStoreCoordinator { get; }
+
+		// @property (copy) NSArray<NSPersistentStoreDescription *> * _Nonnull persistentStoreDescriptions;
+		[Export ("persistentStoreDescriptions", ArgumentSemantic.Copy)]
+		NSPersistentStoreDescription[] PersistentStoreDescriptions { get; set; }
+
+		// -(instancetype _Nonnull)initWithName:(NSString * _Nonnull)name;
+		[Export ("initWithName:")]
+		IntPtr Constructor (string name);
+
+		// -(instancetype _Nonnull)initWithName:(NSString * _Nonnull)name managedObjectModel:(NSManagedObjectModel * _Nonnull)model __attribute__((objc_designated_initializer));
+		[Export ("initWithName:managedObjectModel:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string name, NSManagedObjectModel model);
+
+		// -(void)loadPersistentStoresWithCompletionHandler:(void (^ _Nonnull)(NSPersistentStoreDescription * _Nonnull, NSError * _Nullable))block;
+		[Export ("loadPersistentStoresWithCompletionHandler:")]
+		void LoadPersistentStoresWithCompletionHandler (Action<NSPersistentStoreDescription, NSError> block);
+
+		// -(NSManagedObjectContext * _Nonnull)newBackgroundContext __attribute__((ns_returns_retained));
+		[Export ("newBackgroundContext")]
+		NSManagedObjectContext NewBackgroundContext { get; }
```

Comment:
check doc to see if it should be a method (e.g. every call returning a new instance)